### ISSUE TITLE
Add `PDO::ATTR_STRINGIFY_FETCHES => true` in DbPDO.php

### ISSUE THREAD

**Issue by tleon** [PrestaShop/PrestaShop#36942]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

This config needs to be added back into DbPDO.php . It was supposed to be introduced during a merge PR between `8.2.x` and `develop` but the impact on the merge PR was too big. 

We decided that it would be handled in a separate issue. 

This config will keep all data as strings and stops automatic casting to integers, floats and other types. 

Adding  the config [here](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/db/DbPDO.php#L66) 
``` 
        return new PDO(
            $dsn,
            $user,
            $password,
            [
                PDO::ATTR_TIMEOUT => $timeout,
                PDO::MYSQL_ATTR_USE_BUFFERED_QUERY => true,
                PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8mb4',
                PDO::MYSQL_ATTR_MULTI_STATEMENTS => _PS_ALLOW_MULTI_STATEMENTS_QUERIES_,
                // FIX This option keeps all data as strings and stops automatic casting to integers, floats and other types
                PDO::ATTR_STRINGIFY_FETCHES => true,
            ]
        );
``` 
Will break a lot of CI tests. You will need to go through the errors and force a cast into the correct type where it's needed in order to fix the CI.

### Steps to reproduce

N/A. 

### PrestaShop version(s) where the bug happened

9.0.0

### PHP version(s) where the bug happened

8.1

### Your company or customer's name goes here (if applicable).

PrestaShop SA

**Comment by jolelievre** [comment]
This configuration was integrated in 8.2.x so that we have a homogenous behaviour on this PrestaShop version for all the PHP versions => all data returned by PDO is always a string regardless of the PHP version.

For the 9.0 version since it's a major version it's the right moment to introduce breaking changes, besides the minimum version is now 8.1 so we should follow the language new stricter typing which is already the case on the develpo/9.0.x branches

TLDR; this is not relevant starting from PrestaShop 9.0 we want stricter types everywhere
